compiler/rustc_mir/src/transform/coverage/tests.rs

```diff
@@ -0,0 +1,632 @@
+use super::debug;
+use super::graph;
+
+use coverage_test_macros::let_bcb;
+
+use rustc_data_structures::graph::WithNumNodes;
+use rustc_data_structures::graph::WithSuccessors;
+use rustc_index::vec::{Idx, IndexVec};
+use rustc_middle::mir::*;
+use rustc_middle::ty::{self, DebruijnIndex, TyS, TypeFlags};
+use rustc_span::DUMMY_SP;
+
+fn dummy_ty() -> &'static TyS<'static> {
```

Comment:
@Mark-Simulacrum - Does this look reasonable?

Reply:
lgtm. We do compare `TyS` using pointer comparision, so this is only fine as long as you don't ever allocate two `TyS` for the same underlying type.

Afaict you only allocate a `ty::Bool` once here so this is fine